presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry.py

```diff
@@ -126,67 +93,17 @@ def load_predefined_recognizers(
         :param nlp_engine: The NLP engine to use.
         :return: None
         """
-        if not languages:
-            languages = ["en"]
-
-        recognizers_map = {
-            "en": [
-                UsBankRecognizer,
-                UsLicenseRecognizer,
-                UsItinRecognizer,
-                UsPassportRecognizer,
-                UsSsnRecognizer,
-                NhsRecognizer,
-                SgFinRecognizer,
-                AuAbnRecognizer,
-                AuAcnRecognizer,
-                AuTfnRecognizer,
-                AuMedicareRecognizer,
-                InPanRecognizer,
-                InAadhaarRecognizer,
-                InVehicleRegistrationRecognizer,
-                InVoterRecognizer,
-                InPassportRecognizer,
-            ],
-            "es": [
-                EsNifRecognizer,
-                EsNieRecognizer,
-            ],
-            "it": [
-                ItDriverLicenseRecognizer,
-                ItFiscalCodeRecognizer,
-                ItVatCodeRecognizer,
-                ItIdentityCardRecognizer,
-                ItPassportRecognizer,
-            ],
-            "pl": [PlPeselRecognizer],
-            "ALL": [
-                CreditCardRecognizer,
-                CryptoRecognizer,
-                DateRecognizer,
-                EmailRecognizer,
-                IbanRecognizer,
-                IpRecognizer,
-                MedicalLicenseRecognizer,
-                PhoneRecognizer,
-                UrlRecognizer,
-            ],
-        }
-        for lang in languages:
-            lang_recognizers = [
-                self.__instantiate_recognizer(
-                    recognizer_class=rc, supported_language=lang
-                )
-                for rc in recognizers_map.get(lang, [])
-            ]
-            self.recognizers.extend(lang_recognizers)
-            all_recognizers = [
-                self.__instantiate_recognizer(
-                    recognizer_class=rc, supported_language=lang
-                )
-                for rc in recognizers_map.get("ALL", [])
-            ]
-            self.recognizers.extend(all_recognizers)
+
+        from .recognizer_registry_provider import RecognizerRegistryProvider
+        registry_configuration = {"global_regex_flags": self.global_regex_flags}
+        if languages is not None:
+            registry_configuration["supported_languages"] = languages
+
+        recognizers = RecognizerRegistryProvider(
+            registry_configuration=registry_configuration
+        ).create_recognizer_registry().recognizers
+
+        self.recognizers.extend(recognizers)
```

Comment:
Worth checking for duplications. Isn't the nlp provider added twice?

Reply:
`recognizer_registry` doesn't add the nlp recognizer, so there's no duplication. In fact when the analyzer get initialized without a registry we do the following: 
```
        if not registry:
            logger.info("registry not provided, creating default.")
            provider = RecognizerRegistryProvider(
                registry_configuration={"supported_languages": self.supported_languages}
            )
            registry = provider.create_recognizer_registry()
            registry.add_nlp_recognizer(nlp_engine=self.nlp_engine)
```

Which is basically this function now :D